data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/AggregateResponseEventHandlingStrategy.java

```diff
@@ -6,36 +6,55 @@
 package org.opensearch.dataprepper.plugins.lambda.processor;
 
 import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.event.AggregateEventHandle;
 import org.opensearch.dataprepper.model.event.DefaultEventHandle;
 import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.EventHandle;
 import org.opensearch.dataprepper.model.record.Record;
 import org.opensearch.dataprepper.plugins.lambda.common.ResponseEventHandlingStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.BiConsumer;
 
 public class AggregateResponseEventHandlingStrategy implements ResponseEventHandlingStrategy {
 
+    private static final Logger LOG = LoggerFactory.getLogger(AggregateResponseEventHandlingStrategy.class);
+
     @Override
     public List<Record<Event>> handleEvents(List<Event> parsedEvents, List<Record<Event>> originalRecords,
                                             BiConsumer<Event, Event> consumerUnused) {
 
         List<Record<Event>> resultRecords = new ArrayList<>();
         Event originalEvent = originalRecords.get(0).getData();
-        DefaultEventHandle eventHandle = (DefaultEventHandle) originalEvent.getEventHandle();
-        AcknowledgementSet originalAcknowledgementSet = eventHandle.getAcknowledgementSet();
+        EventHandle handle = originalEvent.getEventHandle();
 
         for (Event responseEvent : parsedEvents) {
-            // Create a new record for the parsed response event
             Record<Event> newRecord = new Record<>(responseEvent);
             resultRecords.add(newRecord);
+        }
 
-            // Add the new event to ack set, older event will be released by core later
-            if (originalAcknowledgementSet != null) {
-                originalAcknowledgementSet.add(responseEvent);
+        if (handle instanceof DefaultEventHandle) {
```

Comment:
I think there is probably a better way to handle this. If we have other `EventHandle` implementations this will fail.

You are doing this in order to call `getAcknowledgementSet()`. Perhaps we should have interfaces for `SingleAcknowledgementSetEventHandle` and `MultipleAcknowledgementSetEventHandle`? Or would a collection of `AcknowledgementSet` on the base `EventHandle` interface be appropriate?

I'm not sure we need to handle this in this PR, but we should consider something more robust.